OZON.ru.js

```diff
@@ -1,396 +1,120 @@
-		var valueText;
-		switch(nameText) {
-		case "От производителя":
-			valueText =
-			valueDOMNode.innerHTML.replace(/^[\s\S]*ANNOTATION\] -->/, "");
-			//Zotero.debug("valueText -> " + valueText);
-			valueText = Zotero.Utilities.cleanTags(valueText);
-			result["abstractNote"] = valueText;
-			break;
-		case "От OZON.ru":
-			valueText =
-			valueDOMNode.textContent.replace(/^[\s\S]*От OZON.ru/, "");
-			valueText = CleanText(valueText);
-			result["edition"] = GetEdition(valueText);
-			break;
-		}
+function getAdditionalProperty(element) {
+	// cyrillic alphabet
+	var matches = /[\u0430-\u044f\u0410-\u042fёЁ]+(?=:)/.exec(element.textContent);
+	if (matches === null) return undefined;
+	var propertyName = matches[0];
+	var propertyValue = ZU.xpathText(element, './a/@title');
+
+	return {
+		name: propertyName,
+		value: propertyValue
 	}
-	return result;
 }
 
-function doWeb(doc, url) {
-	var pdObject = new Object();
-	//pdObject["url"] = url;
-
-	// For the reason I do not understand, OZON.ru may alter its layout
-	// even within the same browser and screen resolution. I only noticed
-	// two layout types so far. Therefore the xpath expressions are
-	// extended with 'or' operator to accommodate both layouts types.
-
-	var pdXPath = '//div[@class="bContentBlock"]/h1|//div[@class="l h1"]/h1';
-	var pdXPathRes = doc.evaluate(
-	pdXPath, doc, null, XPathResult.ANY_TYPE, null);
-	pdObject["title"] = CleanText(pdXPathRes.iterateNext().textContent);
-	pdXPath = '//div[@id="js_basic_properties"]/p|//div[@class="product-detail"]/p';
-	pdXPathRes = doc.evaluate(
-	pdXPath, doc, null, XPathResult.ANY_TYPE, null);
-	pdObject = MergeObjects(pdObject, ParseProductDetail(pdXPathRes));
-
-	var nameXPath = '//div[@class="bTechDescription"]/div/div[2]/span|//div[@class="techDescription"]/div/div[2]/span';
-	var valueXPath = '//div[@class="bTechDescription"]/div/div[3]/span|//div[@class="techDescription"]/div/div[3]/span';
-	var nameXPathRes = doc.evaluate(
-	nameXPath, doc, null, XPathResult.ANY_TYPE, null);
-	var valueXPathRes = doc.evaluate(
-	valueXPath, doc, null, XPathResult.ANY_TYPE, null);
-	pdObject = MergeObjects(
-	pdObject, ParseTechDesc(nameXPathRes, valueXPathRes));
-
-	nameXPath = '//div[@id="detail_description"]/div/table/tbody/tr/td/h3|//div[@id="detail_description"]/table/tbody/tr/td/h3';
-	valueXPath = '//div[@id="detail_description"]/div/table/tbody/tr/td|//div[@id="detail_description"]/table/tbody/tr/td';
-	nameXPathRes = doc.evaluate(
-	nameXPath, doc, null, XPathResult.ANY_TYPE, null);
-	valueXPathRes = doc.evaluate(
-	valueXPath, doc, null, XPathResult.ANY_TYPE, null);
-	pdObject = MergeObjects(
-	pdObject, ParseDetailDesc(nameXPathRes, valueXPathRes));
-
-	var newItem = new Zotero.Item("book");
-	for (var field in pdObject) newItem[field] = pdObject[field];
+function getMultipleCreators(value, creatorType) {
+	return value
+			.split(',')
+			.map(function (d) {
+				return ZU.cleanAuthor(d, creatorType, false);
+			});
```

Comment:
Looks OK in Sublime and Scaffold. That's exactly why I don't like using tabs for indentation :)